acceptance/tests/external_facts/root_uses_default_external_fact_dir.rb

```diff
@@ -4,44 +4,92 @@
 extend Facter::Acceptance::UserFactUtils
 
 #
-# This test is intended to ensure that executable external facts placed into the
-# default facts.d directory are properly found and resolved by Facter when run
-# as root.
+# The first of these tests is intended to ensure that executable external facts placed into the
+# default facts.d directory are properly found and resolved by Facter when run as root.
+# The second ensures that when Facter is given a user specified external fact directory, facts
+# from both it and the default facts.d directory are resolved. Finally, we test that given
+# conflicting external facts in a user specified diretory and the default facts.d directory,
+# the user specified fact wins.
 #
 
-unix_content = <<EOM
+unix_fact_1_content = <<EOM
 #!/bin/sh
-echo "external_fact=testvalue"
+echo "external_fact_1=foo"
 EOM
 
-win_content = <<EOM
+unix_override_content = <<EOM
+#!/bin/sh
+echo "external_fact_1=baz"
+EOM
+
+unix_fact_2_content = <<EOM
+#!/bin/sh
+echo "external_fact_2=bar"
+EOM
+
+win_fact_1_content = <<EOM
+@echo off
+echo external_fact_1=foo
+EOM
+
+win_override_content = <<EOM
+@echo off
+echo external_fact_1=baz
+EOM
+
+win_fact_2_content = <<EOM
 @echo off
-echo external_fact=testvalue
+echo external_fact_2=bar
 EOM
 
 agents.each do |agent|
   os_version = on(agent, facter('kernelmajversion')).stdout.chomp.to_f
   factsd = get_factsd_dir(agent['platform'], os_version)
+  custom_external_dir = get_user_fact_dir(agent['platform'], os_version)
```

Comment:
@MikaelSmith those methods return separate directories based on platform. In *nix we'll get `/opt/puppetlabs/facter/custom` as the `custom_external_dir` and `/opt/puppetlabs/facter/facts.d` as `factsd`